src/runtime/modality.ts

```diff
@@ -10,33 +10,54 @@
 import {assert} from '../platform/assert-web.js';
 
 enum ModalityName {
-  Dom ='dom', DomTouch='dom-touch', Vr='vr', Voice='voice'
+  Dom ='dom', DomTouch='domTouch', Vr='vr', Voice='voice'
 }
 
 export class Modality {
+  // The `names` field in Modality contains the restrictive list of modalities.
+  // `null` stands for no restrictions at all.
+  // empty list stands for no suitable modalities being available.
```

Comment:
It would be clearer if no restrictions were encoded as a separate boolean rather than a null value.

Reply:
done, it is indeed much better, thanks!